### ISSUE TITLE
[HOLD for payment 2023-10-16] [HOLD for payment 2023-10-13] [$500] LOW: Web - Clicking on "Replace" for a paid scan request results in the scan modal opening and then closing automatically

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#27965]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Log in with User A.
2. Open a chat with User B.
3. Go to the "Request money" section and select "Scan."
4. Send a scanned image.
5. Click on the sent image and go to its details.
6. Go to User B's account.
7. Click on "Pay elsewhere."
8. Return to User A's account.
9. Click on the scanned image and attempt to replace it

## Expected Result:
When clicking on "Replace" for a paid scan request, the scan modal should either remain open or not open at all, depending on the intended functionality

## Actual Result:
Clicking on "Replace" for a paid scan request causes the scan modal to open briefly and then close automatically

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.72.8
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/3337595a-de1a-48ed-9b04-dbda16982761

https://github.com/Expensify/App/assets/93399543/f311a33c-b45d-430a-8532-a4a1a4371476

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ayazhussain79
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1695202167946769

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0106bd3efbb87edef7</li>
        <li>Upwork Job ID: 1704962599505321984</li>
        <li>Last Price Increase: 2023-09-21</li>
<li>Automatic offers: </li>
<ul>
<li>dukenv0307 | Contributor | 26918886</li>
</ul></li>
    </ul>
</details>

**Comment by huzaima** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
When clicking on "Replace" for a paid request, the modal should not show the "Replace" option.

### What is the root cause of that problem?
The settled report cannot be edited. It's checked here:
https://github.com/Expensify/App/blob/e24e43124bd008d5dc600de6113aa146d47e486e/src/pages/EditRequestPage.js#L96-L103

Now in `AttachmentModal.js`, we don't check if the request was settled and unconditionally redirect to EditRequestPage. 

### What changes do you think we should make in order to solve the problem?
We should check if the request was settled and shouldn't show the "Replace" menu item if it's settled. In `AttachmentModal.js`, we should check if the parentReport is settled. It's important to check **parentReport**. If we do `ReportUtils.isSettled(props.report.reportID)`, it will be incorrect. We should first get the parent report like this:
```
const parentReport = ReportUtils.getParentReport(props.report);
```
and then check `ReportUtils.isSettled(parentReport.reportID)`.

**Comment by huzaima** [comment]
Contributor details
Your Expensify account email: khanhuzaima@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01d3e07a3afcf2060b

**Comment by dylanexpensify** [comment]
taking over for Puneet as it's part of wave4!

**Comment by dylanexpensify** [comment]
@burczu lets get these proposals reviewed soon! 🙏

**Comment by dylanexpensify** [comment]
@burczu let's get this done today!

**Comment by burczu** [comment]
@dylanexpensify yes, I'm on it right now actually

**Comment by burczu** [comment]
I've just reviewed all the proposals and I believe that all of them recognized the root case correctly. In my opinion removing the `Replace` option from the three dot menu when it's impossible to replace the receipt is the best from the UX point of view. 

[The proposal](https://github.com/Expensify/App/issues/27965#issuecomment-1730733469) from @dukenv0307 presents the solution in the most comprehensive way - I like it the most and would recommend proceeding with it.

However [the proposal](https://github.com/Expensify/App/issues/27965#issuecomment-1730395412)l from @neonbhai (Solution 2) is very similar (but not described that well) and it's hard to tell who was first here - I've checked the editing history and it seems the detailed explanation was added after @dukenv0307.

So, whoever will review these proposals after me, please take it into consideration when making the final decision.

🎀 👀 🎀 C+ reviewed

**Comment by dylanexpensify** [comment]
@jliexpensify lemme know if you can get to this today! Thanks @burczu!!

**Comment by jliexpensify** [comment]
@dylanexpensify i think you meant to tag @Li357 haha!

**Comment by Li357** [comment]
@dukenv0307 Hmm, I see deeplinking for the attachment modal working correctly here:


https://github.com/Expensify/App/assets/12980971/f93cdc22-cab6-4fe1-abb4-d1ff015ffc42

If it's not working now then yes, let's just not show the replace option. If deeplinking is working correctly, let's also fix that with @neonbhai's solution.

**Comment by Li357** [comment]
Ah, I think I was confused on which page was being deeplinked. @neonbhai is suggesting that we show a not found when we deep link to the edit request route, not the modal view. @dukenv0307 it doesn't seem like your proposal handles this case

**Comment by Li357** [comment]
@dukenv0307 I see. Confirmed with Vit internally this is what we're doing and doing something else is out of scope. Assigning you to the issue. Thanks @neonbhai for the patience and proposal as well.

**Comment by dylanexpensify** [comment]
@burczu let's get this reviewed!

**Comment by jliexpensify** [comment]
@dukenv0307 the PR caused a regression right? In that case, we'd have to halve the payment?

**Comment by dylanexpensify** [comment]
Payment summary:

- Reporter: @ayazhussain79 $50
- Contributor: @dukenv0307 $500

**Comment by dylanexpensify** [comment]
reviewing today!

**Comment by burczu** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

* [x]  [@burczu] The PR that introduced the bug has been identified. Link to the PR: This is not regression imo, this situation just wasn't handled before.
* [x]  [@burczu] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: n/a
* [x]  [@burczu] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: n/a
* [x]  [@burczu] Determine if we should create a regression test for this bug.
* [x]  [@burczu] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
* [ ]  [@dylanexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by dylanexpensify** [comment]
payment upcoming!

**Comment by dylanexpensify** [comment]
@ayazhussain79 please apply to job for reporter payment!

**Comment by ayazhussain79** [comment]
@dylanexpensify applied on upwork, Thank you

**Comment by dylanexpensify** [comment]
offer sent @ayazhussain79!

**Comment by dylanexpensify** [comment]
done!
